globus_cli/parsing/shell_completion.py

```diff
@@ -168,12 +186,71 @@ def do_zsh_complete():
         cur = None
         completed_args = comp_words
 
+    def clean_help(helpstr):
+        """
+        Replace ' with '"'"'
+
+        Because we'll put these single quote chars in '...'
+        quotation, we need to do
+        '   -- end single quotes
+        "'" -- single quote string (will concatenate in ZSH)
+        '   -- start single quotes again
+        """
+        return helpstr.replace("'", "'\"'\"'")
```

Comment:
I think *something* needs to be done here for backticks in help strings as well; right now, zsh tries to expand them when you return to it the `_arguments` structure.

The help strings used by the various options to `globus ls` contain a lot of backticks, so `globus ls --<TAB>` spits out errors like

```
(eval):.:1: not enough arguments
(eval):1: command not found: --recursive
```

along with backtick-expanded help output like

```
--long                         -- For text output only. Do long form output, kind of like total 84\ndrwxr-xr-x  4 mayor users  409
```

If we had any help strings with `$`, then that would be similarly problematic (but we don't).

Reply:
I was a little bit nervous about this but thought it was okay. I can confirm that I see the same issue.

I'm a bit confused though -- this is all inside of a single quoted string, so shouldn't backticks and `$` be suppressed? Have I miscounted the number of layers of escaping or something?

Reply:
> I'm a bit confused though -- this is all inside of a single quoted string, so shouldn't backticks and `$` be suppressed? Have I miscounted the number of layers of escaping or something?

I am not really a user of zsh let alone familiar with its internals, but I *think* that whatever you pass to `_arguments` might be re`eval`uated?
https://github.com/zsh-users/zsh/blob/master/Completion/Base/Utility/_arguments#L423

Reply:
I looked at the `_arguments` documentation again ( http://zsh.sourceforge.net/Doc/Release/Completion-System.html ) because that source... nearly made my head explode at first.
After rereading, it looks like that's the processing for descriptions of the form `((item1\:desc1 ...))`. (Since you found this, you've presumably already worked that out -- I'm trying to catch up here and confirm.)

The docs don't make it _at all_ clear that what we're passing is going to be `eval`ed, but that's definitely what's happening.
I've got a fix I'm playing with and trying to break now. I'll push it if I can't find a way of causing harm.